File: csharp/ToolGood.Algorithm/DiyNameInfo.cs
using System.Collections.Generic;

namespace ToolGood.Algorithm
{
    /// <summary>
    /// 自定义类型
    /// </summary>
    public sealed class DiyNameInfo
    {
        /// <summary>
        /// 自定义 参数
        /// </summary>
        public List<string> Parameters { get; private set; }

        /// <summary>
        /// 自定义方法
        /// </summary>
        public List<string> Functions { get; private set; }

        /// <summary>
        /// 自定义类型
        /// </summary>
        public DiyNameInfo()
        {
            Parameters = new List<string>();
            Functions = new List<string>();
        }
    }
}

File: java/src/main/java/toolgood/algorithm/internals/DiyNameInfo.java
/**
 * 自定义类型
 */
package toolgood.algorithm.internals;

import java.util.ArrayList;
import java.util.List;

public final class DiyNameInfo {
    /**
     * 自定义 参数
     */
    private List<DiyNameKeyInfo> parameters;

    /**
     * 自定义方法
     */
    private List<DiyNameKeyInfo> functions;

    /**
     * 自定义类型
     */
    public DiyNameInfo() {
        parameters = new ArrayList<DiyNameKeyInfo>();
        functions = new ArrayList<DiyNameKeyInfo>();
    }

    public List<DiyNameKeyInfo> getParameters() {
        return parameters;
    }

    public List<DiyNameKeyInfo> getFunctions() {
        return functions;
    }
}
